### ISSUE TITLE
Make unable to connect notification dismissable?

### ISSUE THREAD

**Issue by codebam** [siacs/Conversations#1815]
Would it be possible to make this notification dismissable? On networks that block XMPP (like my school) this is a permanent notification and cannot be removed without disabling the account and re-enabling it later.

**Comment by bastlwasti** [comment]
+1 
thats also the complain I get most from my friends about the app.
- or an expert option to opt it out

**Comment by iNPUTmice** [comment]
>  On networks that block XMPP (like my school) 

did you try with conversations.im? That should probably work behind firewalls.

Fixing the problem is a better solution than to provide complicated ways to ignore them.

**Comment by bastlwasti** [comment]
"

    On networks that block XMPP (like my school)

did you try with conversations.im? That should probably work behind firewalls.

Fixing the problem is a better solution than to provide complicated ways to ignore them.
"
There are a lot of hotels, and cafés which allow just web-access, and block all other ports. So the "fixing the problem ..." is not possible, or does  conversations have bosh/websocket feature? will it ever?

**Comment by iNPUTmice** [comment]
> There are a lot of hotels, and cafés which allow just web-access, and block all other ports. So the "fixing the problem ..." is not possible, or does conversations have bosh/websocket feature? will it ever?

conversations.im accepts TLS connections on port 443 which should be - to most firewalls - indistinguishable from HTTPS.

That's why I asked if it works with conversations.im

**Comment by bastlwasti** [comment]
"That's why I asked if it works with conversations.im "
So to get not annoyed by these messages, you suggest switching the server?

**Comment by iNPUTmice** [comment]
> So to get not annoyed by these messages, you suggest switching the server?

No. But you can use this as test case and if it is working enable XEP-0368 on your own server.

**Comment by bastlwasti** [comment]
oh ok, this makes more sense than I thought! ;)

Daniel, to be frankly, I would like to see an option to make this message away, Whatsapp & co also don't have this. But I completely understand that you can't fulfill every wish of any single user preference and wish, so suggest I would download this Android SDK and program this option on my own, would you mind then merging my patch into your source branch?

greets

**Comment by codebam** [comment]
I forgot to check this thread. Tried it, works with conversations.im, but
that isn't really a solution. I'd like to use riseup.net which is blocked
for xmpp. I don't want to disable the account every time I'm somewhere that
blocks it. Can the message just not be permanent and allow it to be
dismissed please?

**Comment by bastlwasti** [comment]
yes, I think everyone would be happy if the option of "go to settings" or "disable account" --> changes to "go to settings" or "don't notify my again". 

I have 30 contacts and every second week I loose one of them because 
1.) they didnt update their prepaid money
2.) their mobile network was too far away in the village
3.) I made an maintenance update at the server

This dialog is not for dummy users, its for experts. The dummy user doesn't know what to do in "go to settings", so they chose the other option "disable account". They are not aware that they can press "back" on their android phone, they feel as if they have to decide between those options.

Last week, I received 20 messages of a friend of mine in the netherlands, he hit "disable account" after a network problem 2 months ago, but was not aware that this may disable the account! :D He noticed the "waiting" state for each message he wrote, but didn't know what this means. After a while he realized that something is wrong, as he didnt get any more messages, but didn't know where to activate it.

In short, its not a good idea having the simple user in mind to make it for them so easy to disable the account. There is no gain in it, the focus should be on allowing the user to be not annoyed by those error notifications.

thank you

**Comment by licaon-kter** [comment]
If an account is disabled a permanent toast sits in that account when you want to chat asking you if you want to activate it, not sure how you could miss it.

**Comment by bastlwasti** [comment]
Ah, you are right for the last example I gave, he got probably an older version.

But the general problem persists: You get this dialog only if you start the app manually, no notifications aren't received when it is disabled.

And: usually the people only have one or two contacts in their Converstation install, so they won't start Conversations manualle to check what happend to the disabled account. They are rather annoyed or have no clue that they disabled it.

**Comment by licaon-kter** [comment]
Dismissed as in `keep retrying`?

Usually dismissed means `don't bother me with this` aka disable problematic account, else the annoying notification will come back in a minute, right?

**Comment by jkufner** [comment]
Keep retrying, but don't show the same error again until successfully connected for some time.

Conversations should significantly increase delay between retries until network status changes to save battery.

**Comment by bastlwasti** [comment]
yes, that's also my point. 
The delay for retries may increase, even checking once a day is sufficient if you battery is really so important... BUT don't make it non-functioning.

**Comment by jkufner** [comment]
> even checking once a day is sufficient

I wouldn't go below once an hour, but aggree with the rest.

**Comment by bastlwasti** [comment]
the only question is, how much the user wants to be notified about the network problem -- while Conversations tries to establish it again. Some users might "enjoy" being notified each time, others are justed annoyed by any notification (by the stupid little app a crazy IT guy persuaded them to install).

So maybe instead of asking "go to settings" or "disable account" (=self destruction ;-) ), the user may be asked "notify me if the problem still persists in one hour" or "don't notify me ever again".

**Comment by jkufner** [comment]
> the only question is, how much the user wants to be notified about the network problem

**Once.**

Conversations should remember (non-persistently), that network is still broken and don't annoy user until gets repaired (or until phone is rebooted or Conversations restarted). Typical scenario for this is useless wifi hotspot, which blocks XMPP, or crashed server.

There is no need for asking to remind user later (or never). If user tries to send a chat message, he will see the connection is not available. If network gets back, Conversation will reconnect automatically. If user wonders why he is unreachable, he will open account settings and see the connection status.

Click the notification to open account settings, or swipe the notification away. No buttons needed.

**Comment by tmolitor-stud-tu** [comment]
Please add an expert setting to deactivate this error notification entirely.

Some of my friends are rather annoyed by this and told me they want to uninstall Conversations because of this notifications.
(they are happening quite often for them)

**Comment by bastlwasti** [comment]
thats also my problem.

**Comment by tmolitor-stud-tu** [comment]
@bastlwasti see this one https://github.com/siacs/Conversations/issues/2126
